src/main/java/com/gmail/woosay333/onlinebookstore/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.gmail.woosay333.onlinebookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstFieldName;
+    private String secondFieldName;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.firstFieldName = constraintAnnotation.first();
+        this.secondFieldName = constraintAnnotation.second();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        try {
+            Object firstValue = new BeanWrapperImpl(value).getPropertyValue(firstFieldName);
```

Comment:
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/BeanWrapperImpl.html
> NOTE: As of Spring 2.5, this is - for almost all purposes - an internal class. It is just public in order to allow for access from other framework packages. For standard application access purposes, use the [PropertyAccessorFactory.forBeanPropertyAccess(java.lang.Object)](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/PropertyAccessorFactory.html#forBeanPropertyAccess(java.lang.Object)) factory method instead.

First time seeing this class, so not sure how it works and what it does, had to google it and found out that it is not recommended. overall we could have used simple reflection by getting the class of `value` var and getting a field by `class.getDeclaredField(field)` to then compare the values of those 2 fields.